replication/event.go

```diff
@@ -278,24 +278,42 @@ func (e *QueryEvent) Dump(w io.Writer) {
 }
 
 type GTIDEvent struct {
-	CommitFlag uint8
-	SID        []byte
-	GNO        int64
+	CommitFlag     uint8
+	SID            []byte
+	GNO            int64
+	LastCommitted  int64
+	SequenceNumber int64
 }
 
 func (e *GTIDEvent) Decode(data []byte) error {
-	e.CommitFlag = uint8(data[0])
+	pos := 0
+	e.CommitFlag = uint8(data[pos])
+	pos++
 
-	e.SID = data[1:17]
+	sidLength := 16
+	e.SID = data[pos : pos+sidLength]
+	pos += sidLength
 
-	e.GNO = int64(binary.LittleEndian.Uint64(data[17:]))
+	e.GNO = int64(binary.LittleEndian.Uint64(data[pos:]))
+	pos += 8
+
+	if len(data) >= 42 {
+		if uint8(data[pos]) == 2 {
```

Comment:
what's 2 here mean?

Reply:
In MySQL's Source code . The  definition of Gtid_event give the explanation:
```
class Gtid_event: public Binary_log_event
{
public:
  /*
    The transaction's logical timestamps used for MTS: see
    Transaction_ctx::last_committed and
    Transaction_ctx::sequence_number for details.
    Note: Transaction_ctx is in the MySQL server code.
  */
 ........
  /// Length of typecode for logical timestamps.
  static const int LOGICAL_TIMESTAMP_TYPECODE_LENGTH= 1;
  /// Length of two logical timestamps.
  static const int LOGICAL_TIMESTAMP_LENGTH= 16;
  // Type code used before the logical timestamps.
  static const int LOGICAL_TIMESTAMP_TYPECODE= 2;
  gtid_info gtid_info_struct;
..... 
``` 
LOGICAL_TIMESTAMP_TYPECODE = 2 .

Reply:
should define a const var for 2

Reply:
That's really good . I will do it later.